docs/hashing.md

```diff
@@ -0,0 +1,79 @@
+# Hashing
+
+## Hash Method Generation
+
+:::{warning}
+The overarching theme is to never set the `@attrs.define(unsafe_hash=X)` parameter yourself.
+Leave it at `None` which means that *attrs* will do the right thing for you, depending on the other parameters:
+
+- If you want to make objects hashable by value: use `@define(frozen=True)`.
+- If you want hashing and equality by object identity: use `@define(eq=False)`
+
+Setting `unsafe_hash` yourself can have unexpected consequences so we recommend to tinker with it only if you know exactly what you're doing.
+:::
+
+Under certain circumstances, it's necessary for objects to be *hashable*.
+For example if you want to put them into a {class}`set` or if you want to use them as keys in a {class}`dict`.
+
+The *hash* of an object is an integer that represents the contents of an object.
+It can be obtained by calling `hash` on an object and is implemented by writing a `__hash__` method for your class.
+
+*attrs* will happily write a `__hash__` method for you [^fn1], however it will *not* do so by default.
+Because according to the [definition](https://docs.python.org/3/glossary.html#term-hashable) from the official Python docs, the returned hash has to fulfill certain constraints:
+
+[^fn1]: The hash is computed by hashing a tuple that consists of a unique id for the class plus all attribute values.
+
+1. Two objects that are equal, **must** have the same hash.
+   This means that if `x == y`, it *must* follow that `hash(x) == hash(y)`.
+
+   By default, Python classes are compared *and* hashed by their `id`.
+   That means that every instance of a class has a different hash, no matter what attributes it carries.
+
+   It follows that the moment you (or *attrs*) change the way equality is handled by implementing `__eq__` which is based on attribute values, this constraint is broken.
+   For that reason Python 3 will make a class that has customized equality unhashable.
+   Python 2 on the other hand will happily let you shoot your foot off.
+   Unfortunately, *attrs* still mimics (otherwise unsupported) Python 2's behavior for backward compatibility reasons if you set `hash=False`.
```

Comment:
You mention the `hash` parameter here and line 58, but `unsafe_hash` in a couple other places? Maybe you should consistently use one or the other? Or maybe mention that they are equivalent?

Reply:
ugh right thanks; i thought i caught them all!